src/core/coerce.c

```diff
@@ -255,25 +255,14 @@ MVMString * MVM_coerce_u_s(MVMThreadContext *tc, MVMuint64 i) {
 }
 
 MVMString * MVM_coerce_n_s(MVMThreadContext *tc, MVMnum64 n) {
-    if (n == MVM_num_posinf(tc)) {
-        return MVM_string_ascii_decode_nt(tc, tc->instance->VMString, "Inf");
-    }
-    else if (n == MVM_num_neginf(tc)) {
-        return MVM_string_ascii_decode_nt(tc, tc->instance->VMString, "-Inf");
-    }
-    else if (n != n) {
-        return MVM_string_ascii_decode_nt(tc, tc->instance->VMString, "NaN");
-    }
+    char buf[64];
+    int len = dtoa_grisu3(n, buf, sizeof(buf));
+    if (len < 0)
```

Comment:
Is this even possible? Are there any floats that grisu3 + snprintf-as-fallback can't stringify?

Reply:
I can't see how it's possible, given the size of buffer provided, and what the formatting string implies. But previously the code was assuming that it needed to handle the documented error return value, so I didn't want to change that. There's probably going to be some-or-other static linting tool that will get grumpy if it's unhandled, so I thought it easier to leave it there.